lib/mayaUsd/ufe/UsdUndoConnectionCommands.cpp

```diff
@@ -0,0 +1,351 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "UsdUndoConnectionCommands.h"
+#include "private/Utils.h"
+
+#include <mayaUsd/ufe/Global.h>
+#include <mayaUsd/ufe/UsdAttribute.h>
+#include <mayaUsd/ufe/UsdAttributes.h>
+#include <mayaUsd/ufe/UsdConnectionHandler.h>
+#include <mayaUsd/ufe/UsdConnections.h>
+#include <mayaUsd/ufe/UsdHierarchyHandler.h>
+#include <mayaUsd/ufe/UsdSceneItem.h>
+#include <mayaUsd/ufe/Utils.h>
+#include <mayaUsd/undo/UsdUndoBlock.h>
+#include <mayaUsdUtils/util.h>
+
+#include <pxr/base/tf/diagnostic.h>
+#include <pxr/usd/sdr/registry.h>
+
+#include <ufe/pathString.h>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+namespace {
+
+Ufe::Attribute::Ptr attrFromUfeAttrInfo(const Ufe::AttributeInfo& attrInfo)
+{
+    auto item
+        = std::dynamic_pointer_cast<UsdSceneItem>(Ufe::Hierarchy::createItem(attrInfo.path()));
+    if (!item) {
+        TF_RUNTIME_ERROR("Invalid scene item.");
+        return nullptr;
+    }
+    return UsdAttributes(item).attribute(attrInfo.name());
+}
+
+UsdAttribute* usdAttrFromUfeAttr(const Ufe::Attribute::Ptr& attr)
```

Comment:
Code copied from UsdConnectionHandler.cpp with small adaptation to return the connection result.